// carbon-components-react/lib/components/DataTable/stories/expansion/DataTable-expansion-story.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.BatchExpansion = exports.TallExpansion = exports.ShortExpansion = exports.CompactExpansion = exports.Usage = exports.default = void 0;

require("./DataTable-expansion-story.scss");

var _addonActions = require("@storybook/addon-actions");

var _react = _interopRequireDefault(require("react"));

var _Button = _interopRequireDefault(require("../../../Button"));

var _DataTable = _interopRequireWildcard(require("../../../DataTable"));

var _shared = require("../shared");

var _DataTable2 = _interopRequireDefault(require("../../DataTable.mdx"));

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var _default = {
  title: 'Components/DataTable/Expansion',
  component: _DataTable.default,
  subcomponents: {
    TableExpandHeader: _DataTable.TableExpandHeader,
    TableExpandRow: _DataTable.TableExpandRow,
    TableExpandedRow: _DataTable.TableExpandedRow,
    Table: _DataTable.Table,
    TableBody: _DataTable.TableBody,
    TableCell: _DataTable.TableCell,
    TableContainer: _DataTable.TableContainer,
    TableHead: _DataTable.TableHead,
    TableHeader: _DataTable.TableHeader,
    TableRow: _DataTable.TableRow
  },
  parameters: {
    docs: {
      page: _DataTable2.default
    }
  }
};
exports.default = _default;

var Usage = function Usage() {
  return /*#__PURE__*/_react.default.createElement(_DataTable.default, {
    rows: _shared.rows,
    headers: _shared.headers
  }, function (_ref) {
    var rows = _ref.rows,
        headers = _ref.headers,
        getHeaderProps = _ref.getHeaderProps,
        getRowProps = _ref.getRowProps,
        getTableProps = _ref.getTableProps,
        getTableContainerProps = _ref.getTableContainerProps;
    return /*#__PURE__*/_react.default.createElement(_DataTable.TableContainer, _extends({
      title: "DataTable",
      description: "With expansion"
    }, getTableContainerProps()), /*#__PURE__*/_react.default.createElement(_DataTable.Table, getTableProps(), /*#__PURE__*/_react.default.createElement(_DataTable.TableHead, null, /*#__PURE__*/_react.default.createElement(_DataTable.TableRow, null, /*#__PURE__*/_react.default.createElement(_DataTable.TableExpandHeader, null), headers.map(function (header, i) {
      return /*#__PURE__*/_react.default.createElement(_DataTable.TableHeader, _extends({
        key: i
      }, getHeaderProps({
        header: header
      })), header.header);
    }))), /*#__PURE__*/_react.default.createElement(_DataTable.TableBody, null, rows.map(function (row) {
      return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, {
        key: row.id
      }, /*#__PURE__*/_react.default.createElement(_DataTable.TableExpandRow, getRowProps({
        row: row
      }), row.cells.map(function (cell) {
        return /*#__PURE__*/_react.default.createElement(_DataTable.TableCell, {
          key: cell.id
        }, cell.value);
      })), /*#__PURE__*/_react.default.createElement(_DataTable.TableExpandedRow, {
        colSpan: headers.length + 1,
        className: "demo-expanded-td"
      }, /*#__PURE__*/_react.default.createElement("h6", null, "Expandable row content"), /*#__PURE__*/_react.default.createElement("div", null, "Description here")));
    }))));
  });
};

exports.Usage = Usage;

var CompactExpansion = function CompactExpansion() {
  return /*#__PURE__*/_react.default.createElement(_DataTable.default, {
    rows: _shared.rows,
    headers: _shared.headers
  }, function (_ref2) {
    var rows = _ref2.rows,
        headers = _ref2.headers,
        getHeaderProps = _ref2.getHeaderProps,
        getRowProps = _ref2.getRowProps,
        getTableProps = _ref2.getTableProps,
        onInputChange = _ref2.onInputChange,
        getToolbarProps = _ref2.getToolbarProps,
        getTableContainerProps = _ref2.getTableContainerProps;
    return /*#__PURE__*/_react.default.createElement(_DataTable.TableContainer, _extends({
      title: "DataTable",
      description: "With expansion"
    }, getTableContainerProps()), /*#__PURE__*/_react.default.createElement(_DataTable.TableToolbar, _extends({}, getToolbarProps(), {
      size: "small"
    }), /*#__PURE__*/_react.default.createElement(_DataTable.TableToolbarContent, null, /*#__PURE__*/_react.default.createElement(_DataTable.TableToolbarSearch, {
      persistent: "true",
      onChange: onInputChange
    }), /*#__PURE__*/_react.default.createElement(_DataTable.TableToolbarMenu, null, /*#__PURE__*/_react.default.createElement(_DataTable.TableToolbarAction, {
      onClick: function onClick() {
        return alert('Alert 1');
      }
    }, "Action 1"), /*#__PURE__*/_react.default.createElement(_DataTable.TableToolbarAction, {
      onClick: function onClick() {
        return alert('Alert 2');
      }
    }, "Action 2"), /*#__PURE__*/_react.default.createElement(_DataTable.TableToolbarAction, {
      onClick: function onClick() {
        return alert('Alert 3');
      }
    }, "Action 3")), /*#__PURE__*/_react.default.createElement(_Button.default, {
      onClick: (0, _addonActions.action)('Add new row'),
      size: "small",
      kind: "primary"
    }, "Add new"))), /*#__PURE__*/_react.default.createElement(_DataTable.Table, _extends({}, getTableProps(), {
      size: "compact"
    }), /*#__PURE__*/_react.default.createElement(_DataTable.TableHead, null, /*#__PURE__*/_react.default.createElement(_DataTable.TableRow, null, /*#__PURE__*/_react.default.createElement(_DataTable.TableExpandHeader, null), headers.map(function (header, i) {
      return /*#__PURE__*/_react.default.createElement(_DataTable.TableHeader, _extends({
        key: i
      }, getHeaderProps({
        header: header
      })), header.header);
    }))), /*#__PURE__*/_react.default.createElement(_DataTable.TableBody, null, rows.map(function (row) {
      return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, {
        key: row.id
      }, /*#__PURE__*/_react.default.createElement(_DataTable.TableExpandRow, getRowProps({
        row: row
      }), row.cells.map(function (cell) {
        return /*#__PURE__*/_react.default.createElement(_DataTable.TableCell, {
          key: cell.id
        }, cell.value);
      })), /*#__PURE__*/_react.default.createElement(_DataTable.TableExpandedRow, {
        colSpan: headers.length + 1,
        className: "demo-expanded-td"
      }, /*#__PURE__*/_react.default.createElement("h6", null, "Expandable row content"), /*#__PURE__*/_react.default.createElement("div", null, "Description here")));
    }))));
  });
};

exports.CompactExpansion = CompactExpansion;

var ShortExpansion = function ShortExpansion() {
  return /*#__PURE__*/_react.default.createElement(_DataTable.default, {
    rows: _shared.rows,
    headers: _shared.headers
  }, function (_ref3) {
    var rows = _ref3.rows,
        headers = _ref3.headers,
        getHeaderProps = _ref3.getHeaderProps,
        getRowProps = _ref3.getRowProps,
        getTableProps = _ref3.getTableProps,
        getTableContainerProps = _ref3.getTableContainerProps;
    return /*#__PURE__*/_react.default.createElement(_DataTable.TableContainer, _extends({
      title: "DataTable",
      description: "With expansion"
    }, getTableContainerProps()), /*#__PURE__*/_react.default.createElement(_DataTable.Table, _extends({}, getTableProps(), {
      size: "short"
    }), /*#__PURE__*/_react.default.createElement(_DataTable.TableHead, null, /*#__PURE__*/_react.default.createElement(_DataTable.TableRow, null, /*#__PURE__*/_react.default.createElement(_DataTable.TableExpandHeader, null), headers.map(function (header, i) {
      return /*#__PURE__*/_react.default.createElement(_DataTable.TableHeader, _extends({
        key: i
      }, getHeaderProps({
        header: header
      })), header.header);
    }))), /*#__PURE__*/_react.default.createElement(_DataTable.TableBody, null, rows.map(function (row) {
      return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, {
        key: row.id
      }, /*#__PURE__*/_react.default.createElement(_DataTable.TableExpandRow, getRowProps({
        row: row
      }), row.cells.map(function (cell) {
        return /*#__PURE__*/_react.default.createElement(_DataTable.TableCell, {
          key: cell.id
        }, cell.value);
      })), /*#__PURE__*/_react.default.createElement(_DataTable.TableExpandedRow, {
        colSpan: headers.length + 1,
        className: "demo-expanded-td"
      }, /*#__PURE__*/_react.default.createElement("h6", null, "Expandable row content"), /*#__PURE__*/_react.default.createElement("div", null, "Description here")));
    }))));
  });
};

exports.ShortExpansion = ShortExpansion;

var TallExpansion = function TallExpansion() {
  return /*#__PURE__*/_react.default.createElement(_DataTable.default, {
    rows: _shared.rows,
    headers: _shared.headers
  }, function (_ref4) {
    var rows = _ref4.rows,
        headers = _ref4.headers,
        getHeaderProps = _ref4.getHeaderProps,
        getRowProps = _ref4.getRowProps,
        getTableProps = _ref4.getTableProps,
        getTableContainerProps = _ref4.getTableContainerProps;
    return /*#__PURE__*/_react.default.createElement(_DataTable.TableContainer, _extends({
      title: "DataTable",
      description: "With expansion"
    }, getTableContainerProps()), /*#__PURE__*/_react.default.createElement(_DataTable.Table, _extends({}, getTableProps(), {
      size: "tall"
    }), /*#__PURE__*/_react.default.createElement(_DataTable.TableHead, null, /*#__PURE__*/_react.default.createElement(_DataTable.TableRow, null, /*#__PURE__*/_react.default.createElement(_DataTable.TableExpandHeader, null), headers.map(function (header, i) {
      return /*#__PURE__*/_react.default.createElement(_DataTable.TableHeader, _extends({
        key: i
      }, getHeaderProps({
        header: header
      })), header.header);
    }))), /*#__PURE__*/_react.default.createElement(_DataTable.TableBody, null, rows.map(function (row) {
      return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, {
        key: row.id
      }, /*#__PURE__*/_react.default.createElement(_DataTable.TableExpandRow, getRowProps({
        row: row
      }), row.cells.map(function (cell) {
        return /*#__PURE__*/_react.default.createElement(_DataTable.TableCell, {
          key: cell.id
        }, cell.value);
      })), /*#__PURE__*/_react.default.createElement(_DataTable.TableExpandedRow, {
        colSpan: headers.length + 1,
        className: "demo-expanded-td"
      }, /*#__PURE__*/_react.default.createElement("h6", null, "Expandable row content"), /*#__PURE__*/_react.default.createElement("div", null, "Description here")));
    }))));
  });
};

exports.TallExpansion = TallExpansion;

var BatchExpansion = function BatchExpansion() {
  return /*#__PURE__*/_react.default.createElement(_DataTable.default, {
    rows: _shared.rows,
    headers: _shared.headers,
    render: function render(_ref5) {
      var rows = _ref5.rows,
          headers = _ref5.headers,
          getHeaderProps = _ref5.getHeaderProps,
          getExpandHeaderProps = _ref5.getExpandHeaderProps,
          getRowProps = _ref5.getRowProps,
          getTableProps = _ref5.getTableProps,
          getTableContainerProps = _ref5.getTableContainerProps;
      return /*#__PURE__*/_react.default.createElement(_DataTable.TableContainer, _extends({
        title: "DataTable",
        description: "With batch expansion"
      }, getTableContainerProps()), /*#__PURE__*/_react.default.createElement(_DataTable.Table, getTableProps(), /*#__PURE__*/_react.default.createElement(_DataTable.TableHead, null, /*#__PURE__*/_react.default.createElement(_DataTable.TableRow, null, /*#__PURE__*/_react.default.createElement(_DataTable.TableExpandHeader, _extends({
        enableExpando: true
      }, getExpandHeaderProps())), headers.map(function (header, i) {
        return /*#__PURE__*/_react.default.createElement(_DataTable.TableHeader, _extends({
          key: i
        }, getHeaderProps({
          header: header
        })), header.header);
      }))), /*#__PURE__*/_react.default.createElement(_DataTable.TableBody, null, rows.map(function (row) {
        return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, {
          key: row.id
        }, /*#__PURE__*/_react.default.createElement(_DataTable.TableExpandRow, getRowProps({
          row: row
        }), row.cells.map(function (cell) {
          return /*#__PURE__*/_react.default.createElement(_DataTable.TableCell, {
            key: cell.id
          }, cell.value);
        })), /*#__PURE__*/_react.default.createElement(_DataTable.TableExpandedRow, {
          colSpan: headers.length + 1,
          className: "demo-expanded-td"
        }, /*#__PURE__*/_react.default.createElement("h6", null, "Expandable row content"), /*#__PURE__*/_react.default.createElement("div", null, "Description here")));
      }))));
    }
  });
};

exports.BatchExpansion = BatchExpansion;
